Branch-SDK/src/io/branch/referral/BranchShortLinkBuilder.java

```diff
@@ -137,10 +149,23 @@ public String getShortUrl() {
      * @param callback A {@link Branch.BranchLinkCreateListener} callback instance that will trigger
      */
     public void generateShortUrl(Branch.BranchLinkCreateListener callback) {
-        super.generateUrl(callback);
+        super.generateUrl(callback, true);
     }
 
-    void generateShortUrl(Branch.BranchLinkCreateListener callback, boolean isFromShareSheet) {
-        super.generateUrl(callback, isFromShareSheet);
+    /**
+     * <p>Configures and requests a short URL to be generated by the Branch servers, via a asynchronous
+     * call; The {@link Branch.BranchLinkCreateListener} is called back with the url when the url is generated.</p>
+     *
+     * @param callback         A {@link Branch.BranchLinkCreateListener} callback instance that will trigger
+     * @param defaultToLongUrl A {@link Boolean} whose value should be false if a  long url is not intended to return on failure.
+     *                         Setting this it false will return NULL for url on failure else a long url.
+     */
+    public void generateShortUrl(Branch.BranchLinkCreateListener callback, Boolean defaultToLongUrl) {
```

Comment:
I believe the `defaultToLongUrl` argument can be just a primitive boolean here. This overhead is probably negligible but consistency would be nice.